packages/@react-stately/dnd/src/useDraggableCollectionState.ts

```diff
@@ -101,10 +105,23 @@ export function useDraggableCollectionState(props: DraggableCollectionOptions):
       }
     },
     endDrag(event) {
+      let {draggingCollectionRef, droppedCollectionRef, droppedTarget} = getDnDState();
+      let isInternalDrop = droppedCollectionRef?.current === draggingCollectionRef?.current;
+      let isInternalFolderDrop = isInternalDrop && !(droppedTarget instanceof HTMLElement) && droppedTarget?.type === 'item' && droppedTarget?.dropPosition === 'on' && !!collection.getItem(droppedTarget.key).childNodes;
```

Comment:
Get rid of isInternalFolderDrop, user should handle internal folder drop via onItemDrop